src/python/pants/notes/2.18.x.md

```diff
@@ -1,349 +1,8 @@
-
-## 2.18.0.dev0 (May 26, 2023)
-
-### New Features
-
-* Support scala_artifact ([#19128](https://github.com/pantsbuild/pants/pull/19128))
-
-* pass "--debug" flag to all helm invocations if requested ([#19093](https://github.com/pantsbuild/pants/pull/19093))
-
-* Enable env vars in `node_build_script` ([#19100](https://github.com/pantsbuild/pants/pull/19100))
-
-### User API Changes
-
-* Remove or bump deprecations ([#19169](https://github.com/pantsbuild/pants/pull/19169))
-
-* Switch to [lambdex].layout = "zip" by default, deprecating the section ([#19122](https://github.com/pantsbuild/pants/pull/19122))
-
-* javascript: Allow nodejs-backed tools to point to a resolve and/or package manager ([#19040](https://github.com/pantsbuild/pants/pull/19040))
-
-* Terraform: Hashes for MacOS binaries have changed ([#19004](https://github.com/pantsbuild/pants/pull/19004))
-
-* Implement layout="zip" for Lambda/GCF, deprecating lambdex ([#19076](https://github.com/pantsbuild/pants/pull/19076))
-
-* Run `terraform init` as part of `check` goal ([#18974](https://github.com/pantsbuild/pants/pull/18974))
-
-* Deprecate [mypy].extra_type_stubs and its lockfile. ([#19084](https://github.com/pantsbuild/pants/pull/19084))
-
-* Remove more 2.18.0.dev0 deprecations (follow up to #19073) ([#19077](https://github.com/pantsbuild/pants/pull/19077))
-
-* Bring the tool `--export` option deprecation forward. ([#19079](https://github.com/pantsbuild/pants/pull/19079))
-
-* Only build and release wheels for Python 3.9. ([#18829](https://github.com/pantsbuild/pants/pull/18829))
-
-* Require pants to run via scie-pants. ([#19075](https://github.com/pantsbuild/pants/pull/19075))
-
-* Some 2.18.0.dev0 deprecations ([#19073](https://github.com/pantsbuild/pants/pull/19073))
-
-* Implement layout="zip" for Lambda/GCF, skipping lambdex ([#19022](https://github.com/pantsbuild/pants/pull/19022))
-
-### Plugin API Changes
-
-* Simplify interpreter-related CI config. ([#19096](https://github.com/pantsbuild/pants/pull/19096))
-
-### Bug Fixes
-
-* If a shebang already exists when adding a preamble, preserve it. ([#19133](https://github.com/pantsbuild/pants/pull/19133))
-
-* Workaround `botocore` bug in S3 URL Handler backend ([#19056](https://github.com/pantsbuild/pants/pull/19056))
-
-* Don't try and garbage collect tempfiles ([#19103](https://github.com/pantsbuild/pants/pull/19103))
-
-* Allow the usage of digits in named cache names ([#19052](https://github.com/pantsbuild/pants/pull/19052))
-
-### Documentation
-
-* Fix docs link to pants_from_sources script. ([#19088](https://github.com/pantsbuild/pants/pull/19088))
-
-* Add `botocore-a-la-carte` reference too `s3` URL Handler ([#19055](https://github.com/pantsbuild/pants/pull/19055))
-
-* add bazel-remote to list of compatible remote caches ([#19041](https://github.com/pantsbuild/pants/pull/19041))
+For the full changelog, see the individual GitHub Releases for this series: https://github.com/pantsbuild/pants/releases
```

Comment:
FWIW, I'd like to improve this, because digging through the many releases to find all the 2.18 ones is very hard, and being able to search a single page to see "oh, is there a change related to <feature/bugfix I'm interested in>" is handy.

It doesn't need to change in this PR, but what do you think about having the stable releases including the full log of all changes across the unstable releases leading up to it, e.g. 2.18.0 will include the changelog for 2.18.0.dev0 through to 2.18.0, and 2.18.1 will include 2.18.1rc0 to 2.18.1, etc.

Reply:
My thought would be we host this on the docs site.

Since it's easy to automate basically anything pre-push, it should be easy to scrape the releases and make a changelog.

I was saving that for the docs port though.

Reply:
Hm, that a bit sounds like keeping a changelog in a file in the repo, to be included as part of the doc site generation. As in, we're reverting that behaviour here and then restoring it... but better, the second time, I guess?

(I'm making the assumption we wouldn't be doing the scraping every time we build/deploy the doc site.)

So I understand, do you have thoughts on what goes in the stable releases on Github?

Reply:
No we very much would be scraping for every deployment.

We already generate the reference docs, this can be generated too 🙂

Reply:
Hm, the reference docs feel like a different category of generation, as they're deterministic from the checked in sources in-repo, while this would be hitting an external service that may require auth etc. (As in, using cacheable `adhoc_tool` or other codegen target is fine for reference docs, but wouldn't be fine for changelog API queries.)

As in, I don't feel "we generate reference docs from source code as part of docs build" is a strong precedent to allow "hit external APIs as part of docs build".

I'm assuming that being able to generate the docs easily locally would be very handy, once we move to sphinx or mkdocs (or whatever), so I'm wanting to get a sense of a plan,  so we don't accidentally go down a dead-end.

Reply:
I was imagining that local builds had varying levels of generation, to allow for fast, or full, local builds.

Fastest would be no generation at all. Then next would be reference docs. Last, and most complete would be changelog.

Also sprinkle in whatever else we'd want to generate as part of docs build.

Regardless though, whatever we come up with. It wouldn't likely be in (this) repo.